python/lsst/ts/m2/model.py

```diff
@@ -0,0 +1,849 @@
+
+    def check_axial_force_limit(self, apply=None):
+        """Check if axial forces are out of bounds.
+
+        Parameters
+        ----------
+        apply : `numpy.ndarray` or `None`, optional
+            Forces to apply in Newton. If `None` use current setup. (default is
+            None.)
+
+        Returns
+        -------
+        `numpy.ndarray`
+            Array with the combined total force per actuator in Newton.
+
+        Raises
+        ------
+        RuntimeError
+            If force limit is out of range.
+        """
+
+        demanded_axial_force = (
+            (apply if apply is not None else self.axial_forces["applied"])
+            + self.axial_forces["lutGravity"]
+            + self.axial_forces["lutTemperature"]
+            + self.axial_forces["hardpointCorrection"]
+        )
+
+        if np.any(demanded_axial_force > self.max_axial_force):
+            actuators = np.where(demanded_axial_force > self.max_axial_force)[0]
+            raise RuntimeError(
+                f"Maximum axial force limit [{self.max_axial_force}N] reached in actuators {actuators}."
+            )
+
+        return demanded_axial_force
+
+    def check_tangent_force_limit(self, apply=None):
+        """Check if tangent forces are out of bounds.
+
+        Parameters
+        ----------
+        apply : `numpy.ndarray` or `None`, optional
+            Forces to apply in Newton. If `None` use current setup. (default is
+            None.)
+
+        Returns
+        -------
+        `numpy.ndarray`
+            Array with the combined total force per actuator in Newton.
+
+        Raises
+        ------
+        RuntimeError
+            If force limit is out of range.
+        """
+
+        demanded_tanget_force = np.array(
+            (apply if apply is not None else self.tangent_forces["applied"])
+            + self.tangent_forces["lutGravity"]
+            + self.tangent_forces["lutTemperature"]
+            + self.tangent_forces["hardpointCorrection"]
+        )
+
+        if np.any(demanded_tanget_force > self.max_tangent_force):
+            actuators = np.where(demanded_tanget_force > self.max_tangent_force)[0]
+            raise RuntimeError(
+                f"Maximum axial force limit [{self.max_tangent_force}N] reached in actuators {actuators}."
+            )
+
+        return demanded_tanget_force
+
+    def reset_force_offsets(self):
+        """Reset the force offsets."""
+
+        self.axial_forces["applied"] = np.zeros(self.n_actuators)
+        self.tangent_forces["applied"] = np.zeros(self.n_tangent_actuators)
+
+    def clear_errors(self):
+        """Clear the errors."""
+
+        self.error_cleared = True
+
+    def select_inclination_source(self, source):
+        """Select the inclination source.
+
+        Parameters
+        ----------
+        source : `int`
+            Inclination source based on the enum:
+            lsst.ts.idl.enums.MTM2.InclinationTelemetrySource.
```

Comment:
You can probably ditch the  `lsst.ts.idl.enums.` part.

Reply:
Done in the commit: f20e3d7. Thanks!